### ISSUE TITLE
cmd/link: segmentation fault when compiling plugin

### ISSUE THREAD

**Issue by nuss-justin** [golang/go#17139]
Please answer these questions before submitting your issue. Thanks!
#### What did you do?

Tried to compile the following:

``` go
package main

// // Nothing to do
import "C"

func init() {
}

func main() {

}
```

Command: `go build -buildmode=plugin plugin.go`
#### What did you expect to see?

No errors.
#### What did you see instead?

```
# command-line-arguments
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xd8 pc=0x47abaa]

goroutine 1 [running]:
panic(0x5a2480, 0x68f4c0)
    /opt/go/src/runtime/panic.go:496 +0x1a0
cmd/link/internal/ld.(*deadcodepass).init(0xc420043ce8)
    /opt/go/src/cmd/link/internal/ld/deadcode.go:253 +0x6ba
cmd/link/internal/ld.deadcode(0xc4204c0000)
    /opt/go/src/cmd/link/internal/ld/deadcode.go:59 +0xca
cmd/link/internal/ld.Main()
    /opt/go/src/cmd/link/internal/ld/main.go:185 +0x802
main.main()
    /opt/go/src/cmd/link/main.go:55 +0x142
```
#### System details

```
go version devel +2e2db7a Sat Sep 17 02:54:11 2016 +0000 linux/amd64
GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/justinn/Workspace/go"
GORACE=""
GOROOT="/opt/go"
GOTOOLDIR="/opt/go/pkg/tool/linux_amd64"
TERM="dumb"
CC="gcc"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build787786251=/tmp/go-build -gno-record-gcc-switches"
CXX="g++"
CGO_ENABLED="1"
uname -sr: Linux 4.7.2-1-ARCH
/usr/lib/libc.so.6: GNU C Library (GNU libc) stable release version 2.24, by Roland McGrath et al.
gdb --version: GNU gdb (GDB) 7.11.1
```

**Comment by rogpeppe** [comment]
This seems to be the case when there are no exported symbols.
I get the same error when compiling the program

    package main

**Comment by crawshaw** [comment]
Fixed by 8eb9fda.
